java/arcs/core/data/Capabilities.kt

```diff
@@ -50,5 +50,18 @@ data class Capabilities(val capabilities: Set<Capability>) {
         )
         val TiedToRuntime: Capabilities = Capabilities(setOf<Capability>(Capability.TiedToRuntime))
         val TiedToArc: Capabilities = Capabilities(setOf<Capability>(Capability.TiedToArc))
+
+        fun fromAnnotations(annotations: List<Annotation>): Capabilities {
+            val capabilities = mutableSetOf<Capability>()
+            annotations.forEach {
+                when (it.name) {
+                    "persistent" -> capabilities.add(Capability.Persistent)
```

Comment:
This could be rewritten as Capability.valueOf(it.name.decapitalize()) instead of using when

Reply:
Actually I see you want to silently ignore unknown capabilities? If so, you'd need to trap the IllegalArgumentException and ignore it, and ignore my test suggestion below.

Reply:
annotations can contain other things, not just capabilities, so i don't want to use `valueOf` and have it throw.
I added a non-capability annotation to the test instead.